lib/auth/sessions.go

```diff
@@ -388,3 +463,44 @@ func (a *Server) CreateSAMLIdPSession(ctx context.Context, req types.CreateSAMLI
 
 	return session, nil
 }
+
+type CreateAppSessionForV15Client interface {
+	Ping(ctx context.Context) (proto.PingResponse, error)
+	CreateAppSession(ctx context.Context, req *proto.CreateAppSessionRequest) (types.WebSession, error)
+	AppSessionWatcher
+}
+
+// TryCreateAppSessionForClientCertV15 creates an app session if the auth
+// server is pre-v16 and returns the app session ID. This app session ID
+// is needed for user app certs requests before v16.
+// TODO (Joerger): DELETE IN v17.0.0
+func TryCreateAppSessionForClientCertV15(ctx context.Context, client CreateAppSessionForV15Client, username string, routeToApp proto.RouteToApp) (string, error) {
+	pingResp, err := client.Ping(ctx)
```

Comment:
Do we need the extra round trip? Can we use `metadata.ClientVersionFromContext()` here?

Reply:
This is actually called by the client to check the server version, so we need the roundtrip.

Reply:
both `tc` and `tbot` have some cache over ping response. Ideally we can use those, but I see it's quite a hassle to use them here for a temporary change.